src/components/submission/diff-item.vue

```diff
@@ -0,0 +1,226 @@
+<!--
+Copyright 2021 ODK Central Developers
+See the NOTICE file at the top-level directory of this distribution and at
+https://github.com/getodk/central-frontend/blob/master/NOTICE.
+
+This file is part of ODK Central. It is subject to the license terms in
+the LICENSE file found in the top-level directory of this distribution and at
+https://www.apache.org/licenses/LICENSE-2.0. No part of ODK Central,
+including this file, may be copied, modified, propagated, or distributed
+except according to the terms contained in the LICENSE file.
+-->
+<template>
+  <div class="submission-diff-item">
+    <div v-if="visiblePath.length > 0" class="full-path">
+      <template v-for="(field, index) in visiblePath">
+        <span :key="index">
+          <template v-if="Array.isArray(field)">
+            <span>{{ field[0] }}</span>
+            <span class="field-counter"> #</span><span>{{ +field[1] + 1 }}</span>
+          </template>
+          <template v-else>
+            <span>{{ field }}</span>
+          </template>
+          <template v-if="index < visiblePath.length - 1">
+            <span class="field-separator"> &rsaquo; </span>
+          </template>
+        </span>
+      </template>
+    </div>
+    <div>
+    <div class="diff-details">
+      <template v-if="isAtomicChange">
+        <div class="field-name" :title="fieldName">
+          {{ fieldName }}
+        </div>
+        <div class="old-to-new">
+          <span v-if="entry.old" class="data-old">
+            <span v-if="isBinary">
```

Comment:
Edge case:

- A form is published, and a submission is created.
- A binary field is removed from the form, and a new form version is published.
- The submission is edited.
- The diff will show **filename -> empty**, but filename won't be linked.

This seems very unlikely, and there isn't an easy way to address it, but just wanted to note that we thought about this case.